### PR TITLE
process object tx

### PR DESCRIPTION
**Pull request by dorisRiver**

### CODE REVIEW

File: chain/node.go [hunk 1]
```diff
@@ -56,17 +57,24 @@ func (n *NodeClient) BlockByHash(ctx context.Context, hash string) (*gateway.Blo
...
 	}
 
-	url := fmt.Sprintf("%s/%s?public_key=%s", n.endpoint, account, pk)
+	url := fmt.Sprintf("%s/%s?public_key=%s", n.endpoint, accountPath, pk)
 	a := &gateway.AccountResp{}
 	return a, httpGet(ctx, url, a)
 }
 
+// Storage gets storage contract detail by tx hash.
+func (n *NodeClient) Storage(ctx context.Context, hash string) (*gateway.StorageResp, error) {
```
Comment (maxkaneco) [inline chain/node.go:72]: StorageContract

File: indexer/block.go [hunk 2]
```diff
@@ -77,62 +77,167 @@ func (e *EventProcessor) processTransactions(
...
+		int64(tx.BalanceTransfer.Amount),
+	)
+}
+
+func (e *EventProcessor) processObjectCommitTx(
+	db *gorm.DB,
+	txID uint64,
+	hash string,
+) error {
+	sr, err := e.node.Storage(e.ctx, hash)
```
Comment (maxkaneco) [inline indexer/block.go:131]: sr -> sc

**Review by maxkaneco — COMMENTED** [review]
Looks good overall. Just a few minor comments

File: indexer/block.go [hunk 3]
```diff
@@ -77,62 +77,167 @@ func (e *EventProcessor) processTransactions(
...
-			Hash:     t.TxHash,
-			From:     t.From,
-			Position: uint64(i),
-			GasPrice: t.GasPrice,
-			Type:     pbc.TxType_value[t.Type],
-			Raw:      raw,
-		}
+		switch tx.Type {
+		case pbc.TxType_BALANCE_TRANSFER.String():
+			if err := e.processBalanceTransferTx(db, tx); err != nil {
```
[the remark this answers is not in the document]
Comment (dorisRiver) [reply indexer/block.go]: it‘s in a loop of block transactions

**Review by maxkaneco — APPROVED** [review]

**Comment by maxkaneco** [comment]
PPH=3hrs

**Pull request merged**